app/src/main/res/layout/welcome.xml

```diff
@@ -11,44 +11,49 @@
 
   <androidx.constraintlayout.widget.ConstraintLayout
     android:layout_width="match_parent"
-    android:layout_height="wrap_content">
+    android:layout_height="wrap_content"
+    android:paddingStart="28dp"
+    android:paddingEnd="28dp">
 
-    <TextView
-      android:id="@+id/welcome_text_view"
-      android:layout_width="0dp"
+    <com.google.android.flexbox.FlexboxLayout
+      android:id="@+id/welcome_message_container"
+      android:layout_width="match_parent"
       android:layout_height="wrap_content"
-      android:layout_marginStart="28dp"
-      android:fontFamily="sans-serif"
-      android:textColor="@color/oppiaPrimaryText"
-      android:textSize="24sp"
-      android:text="@{viewModel.greeting}"
+      app:alignContent="flex_start"
+      app:alignItems="flex_start"
+      app:flexDirection="row"
+      app:flexWrap="wrap"
+      app:justifyContent="flex_start"
+      app:layout_constraintEnd_toEndOf="parent"
       app:layout_constraintStart_toStartOf="parent"
-      app:layout_constraintTop_toTopOf="parent" />
+      app:layout_constraintTop_toTopOf="parent">
 
-    <TextView
-      android:id="@+id/profile_name_textview"
-      android:layout_width="0dp"
-      android:layout_height="0dp"
-      android:layout_marginStart="8dp"
-      android:layout_marginEnd="28dp"
-      android:ellipsize="end"
-      android:maxLines="1"
-      android:fontFamily="sans-serif"
-      android:text="@{String.format(@string/welcome_profile_name, viewModel.profileName)}"
-      android:textColor="@color/oppiaPrimaryText"
-      android:textSize="24sp"
-      app:layout_constraintEnd_toEndOf="parent"
-      app:layout_constraintStart_toEndOf="@+id/welcome_text_view"
-      app:layout_constraintBottom_toBottomOf="@id/welcome_text_view"
-      app:layout_constraintTop_toTopOf="@+id/welcome_text_view" />
+      <TextView
+        android:id="@+id/welcome_text_view"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:layout_marginEnd="4dp"
+        android:fontFamily="sans-serif"
+        android:text="@{viewModel.greeting}"
+        android:textColor="@color/oppiaPrimaryText"
+        android:textSize="24sp" />
+
+      <TextView
+        android:id="@+id/profile_name_textview"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:fontFamily="sans-serif"
+        android:text="@{String.format(@string/welcome_profile_name, viewModel.profileName)}"
+        android:textColor="@color/oppiaPrimaryText"
+        android:textSize="24sp" />
+    </com.google.android.flexbox.FlexboxLayout>
 
     <View
-      android:layout_width="0dp"
+      android:layout_width="172dp"
```

Comment:
Could we try setting this to 0dp and constraining the end to the end of the container, then making the container wrap_content & give it horizontal bias to be left-aligned in the constraint layout? I think flexbox layout is supposed to support growing up to its maximum before wrapping.

Reply:
But in that case, the orange line could take the full width of the screen. Do we want that?

Reply:
I am expecting the following output if your suggestion is implemented - 
If the salutation and username appear side to side, then the orange line would be as wide as the `flexboxlayout` and if the username appears below the salutation then the orange line would be as wide as the username's first line.

So, if we want this behaviour, I'll make changes in the next commit.

Reply:
That's up to @mschanteltc & @rt4914. My understanding about why we decided to go with a fixed size was because we couldn't make it as long as the text, but this is an approximation of that even if it might sometimes span the whole length of the view.

FWIW I think the greeting-only line length looks better in the longer name situations.